src/atrule/keyframes.rs

```diff
@@ -0,0 +1,161 @@
+use codemap::{Span, Spanned};
+
+use peekmore::PeekMoreIterator;
+
+use super::parse::ruleset_eval;
+use crate::error::SassResult;
+use crate::scope::Scope;
+use crate::selector::Selector;
+use crate::utils::{devour_whitespace, eat_number};
+use crate::value::Number;
+use crate::{RuleSet, Stmt, Token};
+
+#[derive(Debug, Clone)]
+pub(crate) struct Keyframes {
+    pub super_selector: Selector,
+    pub name: String,
+    pub block_list: Vec<(Vec<Spanned<KeyframeSelector>>, Vec<Spanned<Stmt>>)>,
+}
+
+#[derive(Debug, Clone)]
+pub(crate) enum KeyframeSelector {
```

Comment:
This should probably implement display so that it can be emitted correctly

```rust
match self {
    Self::From => write!(f, "from"),
    Self::To => write!(f, "to"),
    Self::Percentage(v) => write!(f, "{}%", v),
}

Reply:
Ah, I didn't know it is based on `Display`, let me add that.